### PR TITLE
Update review actions and snapshot/build states to reflect renaming in API

### PR DESCRIPTION
**Pull request by cadeParade**
## Description
This PR allows percy-web to work with the new naming of snapshot/build `review_state`, `review_state_reason`, and review `action`.

### build and snapshot:
`review_state: :rejected` to `review_state: :changes_requested`

### build:
`review_state_reason: :rejected_snapshot` to `review_state: :changes_requested_snapshot`
`review_state_reason: :rejected_snapshot_previously` to `review_state: :changes_requested_snapshot_previously`

### snapshot:
`review_state_reason: :user_rejected` to `review_state: :user_requested_changes`
`review_state_reason: :user_rejected_previously` to `review_state: :user_requested_changes_previously`

### review:
`action: :rejected` to `action: :request_changes`

### ADDITIONS:
from `total_snapshots_rejected` to `total_snapshots_requesting_changes`

### CODE REVIEW

File: app/models/build.js [hunk 1]
```diff
@@ -4,6 +4,20 @@ import DS from 'ember-data';
 import moment from 'moment';
 import {countDiffsWithSnapshotsPerBrowser} from 'percy-web/lib/filtered-comparisons';
 
+export const BUILD_APPROVED_STATE = 'approved';
+export const BUILD_UNREVIEWED_STATE = 'unreviewed';
+export const BUILD_REJECTED_STATE = 'changes_requested';
+
+export const BUILD_REVIEW_STATE_REASONS = {
```
Comment (unicornzero) [inline app/models/build.js]: Smart.

File: app/models/build.js [hunk 2]
```diff
@@ -4,6 +4,20 @@ import DS from 'ember-data';
...
+  UNREVIEWED: 'unreviewed_snapshots',
+  ALL_SNAPSHOTS_APPROVED: 'all_snapshots_approved',
+  ALL_SNAPSHOTS_APPROVED_PREVIOUSLY: 'all_snapshots_approved_previously',
+  NO_DIFFS: 'no_diffs',
+  AUTO_APPROVED_BRANCH: 'auto_approved_branch',
+  SNAPSHOT_REJECTED: 'changes_requested_snapshot',
```
Comment (unicornzero) [inline app/models/build.js]: Do you care that it says "rejected" here? I don't know that I care, just asking.

File: app/models/build.js [hunk 3]
```diff
@@ -73,10 +87,10 @@ export default DS.Model.extend({
 
   // Review state, aggregated across all reviews. This will only be set for finished builds.
   reviewState: DS.attr(),
-  isUnreviewed: equal('reviewState', 'unreviewed'),
+  isUnreviewed: equal('reviewState', BUILD_UNREVIEWED_STATE),
+  isApproved: equal('reviewState', BUILD_APPROVED_STATE),
+  isRejected: equal('reviewState', BUILD_REJECTED_STATE),
```
Comment (unicornzero) [inline app/models/build.js]: yeah, "rejected" makes your logic easier to read than "isChangesRequested" or whatnot.

Comment (cadeParade) [reply app/models/build.js]: I agree. I debated changing it all, but I do feel like the tradeoff of having slightly different names is worth the convenience of the simplified language.

**Pull request merged**